src/openms/source/FORMAT/MascotGenericFile.cpp

```diff
@@ -337,7 +338,7 @@ namespace OpenMS
            << "_" << spec.getNativeID() << "_" << filename << "\n";
         os << "PEPMASS=" << precisionWrapper(mz) <<  "\n";
         os << "RTINSECONDS=" << precisionWrapper(rt) << "\n";
-        os << "SCANS=" << spec.getNativeID().substr(spec.getNativeID().find("=")+1) << "\n";
+        os << "SCANS=" << SpectrumLookup::extractScanNumber(spec.getNativeID(), native_id_type_accession) << "\n";
```

Comment:
optimization: SpectrumLookup internally compares a lot of strings and is called for every spectrum.
It might be better to use an enum+switch. But maybe we should benchmark MGF writing first by simply timing FileConverter mgf->mgf